lib/ansible/modules/database/misc/kibana_saved_object.py

```diff
@@ -0,0 +1,446 @@
+          certificates.
+      type: bool
+      default: 'true'
+    client_cert:
+      description:
+        - PEM formatted certificate chain file to be used for SSL client
+          authentication. This file can also include the key as well, and if
+          the key is included, I(client_key) is not required
+    client_key:
+      description:
+        - PEM formatted file that contains your private key to be used for SSL
+          client authentication. If I(client_cert) contains both the certificate
+          and key, this option is not required.
+    searchguard_tenant:
+      type: str
+      description:
+        - Searchguard tenant to use.
+    timeout:
+      type: int
+      description:
+        - Request timeout.
+        - Should be provided in seconds.
+      default: 30
+    type:
+      type: str
+      description:
+        - Type of the saved object.
+      choices:
+        - visualization
+        - dashboard
+        - search
+        - index-pattern
+        - config
+        - timelion-sheet
+    content:
+      type: dict
+      description:
+        - Body of the object.
+        - Should contain "attributes" dictionary.
+        - JSON file can be used like '{{ lookup("template", "content.json.j2") }}'
+        - See L(Kibana saved object API documentation) for details.
+    overwrite:
+      type: bool
+      description:
+        - Overwrite the object.
+      default: false
+'''
+
+EXAMPLES = '''
+- name: Create an index pattern
+  kibana_saved_object:
+    id: my-pattern
+    state: present
+    type: index-pattern
+    kibana_url: http://192.168.0.42:5601/
+    content:
+      attributes:
+        title: my-pattern-*
+
+- name: Create an index pattern with searchguard enabled
+  kibana_saved_object:
+    id: my-pattern
+    state: present
+    type: index-pattern
+    kibana_url: http://192.168.0.42:5601/
+    content:
+      attributes:
+        title: my-pattern-*
+    user: ansible
+    password: s3cr3t
+    force_basic_auth: true
+    searchguard_tenant: my_application
+
+- name: Create an index pattern using a jinja template
+  kibana_saved_object:
+    id: my-pattern
+    state: present
+    type: index-pattern
+    kibana_url: http://192.168.0.42:5601/
+    content: '{{ lookup("template", "content.json.j2") }}'
+    searchguard_tenant: my_application
+'''
+
+RETURN = '''
+msg:
+  description: The result of the operation
+  returned: always
+  type: str
+  sample: 'Object has been created: my-pattern'
+object_id:
+  description: ID of the saved object
+  returned: always
+  type: str
+  sample: 'my-pattern'
+object_type:
+  description: ID of the saved object
+  returned: always
+  type: str
+  sample: 'index-pattern'
+'''
+
+
+import os
+import json
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.urls import fetch_url, url_argument_spec
+
+
+def get_request_params(module, object_id, object_type, kibana_url,
```

Comment:
``timeout`` is passed, but never used
``module`` is passed, but never used

Judging by the actual function, it's a bit confusing. The return data (dict) appears as if it was designed to be used as ``**kwargs`` for the ``fetch_url`` function, but you're accessing the dict directly. As such I would recommend either returning a tuple, and passing the data that way, or using the ``kwargs`` approach.

Reply:
Done.